pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -846,6 +810,80 @@ func builderSecretVolume(bbs corev1alpha1.BuildBuilderSpec) corev1.Volume {
 	}
 }
 
+func setupBindingVolumesAndMounts(bindings []ServiceBinding) ([]corev1.Volume, []corev1.VolumeMount) {
+	volumes := make([]corev1.Volume, 0)
+	volumeMounts := make([]corev1.VolumeMount, 0)
+
+	for _, binding := range bindings {
+		switch b := binding.(type) {
```

Comment:
can this be achieved with the interface instead of a type switch?

Reply:
it could, but i was liking the simplicity of the type switch and not having to have any intermediate service binding types